src/tls_wrap.cc

```diff
@@ -314,6 +315,8 @@ void TLSWrap::EncOut() {
 
 void TLSWrap::EncOutCb(WriteWrap* req_wrap, int status) {
   TLSWrap* wrap = req_wrap->wrap()->Cast<TLSWrap>();
+  req_wrap->~WriteWrap();
+  delete[] reinterpret_cast<char*>(req_wrap);
```

Comment:
I would suggest moving the placement new and delete logic into New and Dispose methods, like I did just now with FSReqWrap in src/node_file.cc.  Having the same logic repeated five times throughout the file is not good software engineering.